src/pid.cpp

```diff
@@ -59,36 +62,67 @@ Pid::Pid(const Pid & source)
   // Copy the realtime buffer to then new PID class
   gains_buffer_ = source.gains_buffer_;
 
+  // Initialize saved i-term values
+  clear_saved_iterm();
+
   // Reset the state of this PID controller
   reset();
 }
 
 Pid::~Pid() {}
 
-void Pid::initPid(double p, double i, double d, double i_max, double i_min, bool antiwindup)
+void Pid::initPid(double p, double i, double d, double i_max, double i_min, bool antiwindup,
+  bool save_iterm)
 {
-  setGains(p, i, d, i_max, i_min, antiwindup);
+  setGains(p, i, d, i_max, i_min, antiwindup, save_iterm);
 
   reset();
 }
 
 void Pid::reset()
 {
+  save_iterm();
+
   p_error_last_ = 0.0;
   p_error_ = 0.0;
   i_error_ = 0.0;
   d_error_ = 0.0;
   cmd_ = 0.0;
 }
 
+void Pid::save_iterm()
+{
+  // If last integral term is less than min_i_term_, just return - don't keep very small terms
+  if (std::fabs(i_term_last_) < min_i_term_) return;
```

Comment:
Is this really necessary? What's the harm of having a very small term? It just seems that the value of min_i_term_ is arbitrary and may not apply to all use cases. Unless you make that a parameter which may be overkill.

Reply:
Agreed - I added this at last minute and was second guessing myself - should either not be there or be a parameter - which I agree is overkill.  Will fix.  Also thought of another improvement - I really don't need extra saved_iterm_ member - that was a legacy of the way it was implemented in RTC with threshold, etc - I can just not reset the Ierror term on reset - and then the "saved" term will also show up in the diag messages (which it doesn't now).  Will fix both of these if you agree.

Reply:
member name is actually i_term_saved_

Reply:
I agree, Make it simpler if possible